### PR TITLE
Add MariaDb1070Platform with native GUID support

### PR DESCRIPTION
**Pull request by delacry**
|      Q       |   A
|------------- | -----------
| Type         | feature
| BC Break     | no

#### Summary

Mariadb 10.7 has new type for storing GUID (UUID), https://mariadb.org/10-7-preview-feature-uuid-data-type/

### CODE REVIEW

File: src/Platforms/MariaDb1070Platform.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace Doctrine\DBAL\Platforms;
+
+use Doctrine\DBAL\Types\Types;
+
+/**
+ * Provides the behavior, features and SQL dialect of the MariaDB 10.2 (10.2.7 GA) database platform.
+ *
+ * Note: Should not be used with versions prior to 10.2.7.
+ */
```
Comment (derrabus) [inline src/Platforms/MariaDb1070Platform.php]: This doc block looks like copy&paste from the 10.2.7 platform. It should be adjusted to match what this new class actually represents.

**Review by derrabus — CHANGES_REQUESTED** [review]
Thank you for your pull request. Since this is a new feature, please target the 3.2.x branch.

**Comment by delacry** [comment]
@derrabus it is a new feature, but when someone uses older doctrine/dbal with newest MariaDB, it won't recognize native guid type, will it be backported also to older versions?

**Comment by derrabus** [comment]
I don't think that would be necessary. If someone uses the latest and greatest MariaDB, it's fair to have them upgrade to the latest and greatest DBAL as well.

**Comment by delacry** [comment]
Why is that PHPUnit failing? I'm sure I fixed that in last commit, and
`Doctrine\DBAL\Tests\Platforms\MariaDb1070PlatformTest::testReturnsJsonTypeDeclarationSQL`
no longer exists.

**Comment by morozov** [comment]
Since this is a type mapping change, we need to make sure that if a user using GUID on PostgreSQL with DBAL 3.1 upgrades to DBAL 3.2, the schema manager will be able to migrate the schema automatically. If not, we'll need to at least document the migration path.

@Rixafy please test the DBAL upgrade and document the current behavior in the comments.

Additionally, it looks like we don't use MariaDB 10.7 on CI: https://github.com/doctrine/dbal/blob/0edd213bd01a4cf79d4ddc5f63622fa6e2149fe0/.github/workflows/continuous-integration.yml#L211-L215

It has to be added to the matrix to have the new mapping integration-tested.

**Comment by delacry** [comment]
@morozov I don't get that note with postgres, do you mean mariadb and migration for converting from byte(16) to uuid? And also unfortunately I'm unable to test it in my projects, I'm using Nette Framework with [Nettrine](https://github.com/nettrine) extensions and it's locked to dbal 2.x versions because of migrations, I would need to update that integrations as well and it would take some time, or made stand-alone project or repo using symfony and I'm little bit busy right now.

But I guess it may fail by default, or when there are all FK from another tables dropped before changing id type, it may work.

**Comment by morozov** [comment]
> I don't get that note with postgres, do you mean mariadb and migration for converting from byte(16) to uuid?

Yes, MariaDB, sorry.

> And also unfortunately I'm unable to test it in my projects […]

You don't have to test it in your project. You can create a table with a GUID field on the previous version of the DBAL, then update the dependency and try to migrate the schema.

> But I guess it may fail by default, or when there are all FK from another tables dropped before changing id type, it may work.

Good point on foreign keys. I'd start simple with just a single column and then test the FK scenario.

**Comment by fd6130** [comment]
Hi, does the latest dbal support mariadb 10.7 uuid type? Or currently still have to use it as binary type?

**Comment by delacry** [comment]
> Hi, does the latest dbal support mariadb 10.7 uuid type? Or currently still have to use it as binary type?

Still binary as far as I know, I added the support but I didn't have time to test it, since I'm not even using dbal 3.

**Pull request closed without merging**